### PR TITLE
(124) Finalise submission (Frontend changes)

### PR DESCRIPTION
**Pull request by esteedqueen**
# Description
Finalises submission and task completion process

# Status
Ready for review

### CODE REVIEW

**Comment by esteedqueen** [comment]
[Finalise submission](https://trello.com/c/NrlDJrP0/124-finalise-submission)

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -14,7 +13,14 @@ def new
   end
 
   def create
-    render 'reviews/create'
+    mark_task_as_completed
```
Comment (leeky) [inline app/controllers/reviews_controller.rb]: Doesn't this produce a potential race condition - i.e. should `mark_task_as_completed` be moved into the happy path of `if submission.save` ?

File: app/models/api/task.rb [hunk 2]
```diff
@@ -1,5 +1,8 @@
 module API
   class Task < Base
     has_one :framework
+
+    # GET /tasks/:id/complete
```
Comment (leeky) [inline app/models/api/task.rb]: Small typo - comment says `GET`, but endpoint is defined as a `POST`

**Comment by leeky** [comment]
@esteedqueen Looks like there's some conflicts, since merging #14. Could you rebase this and take a look at my comments.

Nice work refactoring the tests as well - those stubs were getting rather messy! 🎉

File: app/models/api/task.rb (hunk 2 above)
Comment (esteedqueen) [reply app/models/api/task.rb]: Good spot! 👍

File: app/controllers/reviews_controller.rb (hunk 1 above)
Comment (esteedqueen) [reply app/controllers/reviews_controller.rb]: Ah, that's true, good spot! Fixed.

**Comment by leeky** [comment]
LGTM! :shipit:

**Pull request merged**